src/lang/queryAst.ts

```diff
@@ -792,3 +795,43 @@ export function getObjExprProperty(
   if (index === -1) return null
   return { expr: node.properties[index].value, index }
 }
+
+/**
+ * Given an AST, returns the settings annotation object if it exists.
+ */
+export function getSettingsAnnotation(
+  node?: Node<Program>
+): KclSettingsAnnotation {
+  const settings: KclSettingsAnnotation = {}
```

Comment:
Do you need to distinguish the case when settings are not set in the KCL? Do you care whether it's the default `mm` vs. not set?

Reply:
I would love that yes

Reply:
I made the WASM wrapper function return null if settings aren't found. But it won't distinguish between having the length set in the KCL and not angle, for example. Either it finds settings or it doesn't.